MotionPhotoReader/app/src/main/java/com/google/android/libraries/motionphotoreader/BufferProcessor.java

```diff
@@ -0,0 +1,128 @@
+package com.google.android.libraries.motionphotoreader;
+
+import android.media.MediaCodec;
+import android.media.MediaExtractor;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
+import android.os.Message;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import static android.os.Build.VERSION_CODES.LOLLIPOP;
+
+/**
+ * A handler meant specifically for handling messages received from a motion photo reader.
+ */
+class BufferProcessor {
+    private static final long TIMEOUT_US = 1000L;
+    private static final long US_TO_NS = 1000L;
+
+    /**
+     * Fields shared with motion photo reader.
+     */
+    private final MediaExtractor lowResExtractor;
+    private final MediaCodec lowResDecoder;
+    private final BlockingQueue<Integer> availableInputBuffers;
+    private final BlockingQueue<Bundle> availableOutputBuffers;
+
+    public BufferProcessor(MediaExtractor lowResExtractor, MediaCodec lowResDecoder,
+                           BlockingQueue<Integer> availableInputBuffers, BlockingQueue<Bundle> availableOutputBuffers) {
```

Comment:
nit: since these are a lot of arguments, consider making this one argument per line, i.e.:

```
public BufferProcessor(MediaExtractor lowResExtractor,
   MediaCodec lowResDecoder,
   BlockingQueue<Integer> availableInputBuffers,
   BlockingQueue<Bundle> availalbeOutputBuffers) {
```

Reasoning is that it makes it easier to see the arguments and not need to visually hunt for the commas, and also makes it nicer for merging in new arguments that get added from other developers